### PR TITLE
chore(rln): tests and benchmarks review

### PR DESCRIPTION
**Pull request by seemenkina**
Part of #237 

This PR includes:

- create benchmark from this test https://github.com/vacp2p/zerokit/blob/0521c7349e7aa36e25b43e4663def226451a7147/rln/tests/poseidon_tree.rs#L13

- add additional measurements for `FullMerkleTree` and `OptimalMerkleTree`  using `set_range` 

- extract common logic in tests/ffi.rs 
- refactor tests and benches

### CODE REVIEW

File: rln/tests/poseidon_tree.rs [hunk 1]
```diff
@@ -9,43 +9,20 @@ mod test {
     use utils::{FullMerkleTree, OptimalMerkleTree, ZerokitMerkleProof, ZerokitMerkleTree};
 
     #[test]
-    /// A basic performance comparison between the two supported Merkle Tree implementations
-    fn test_zerokit_merkle_implementations_performances() {
-        use std::time::{Duration, Instant};
-
-        let tree_height = 20;
+    /// The test is checked correctness for `FullMerkleTree` and `OptimalMerkleTree` with Poseidon hash
+    fn test_zerokit_merkle_implementations() {
```
Comment (seemenkina) [inline rln/tests/poseidon_tree.rs:13]: I've only left the test part here.

File: rln/benches/poseidon_tree_benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    hashers::PoseidonHash,
+};
+use utils::{FullMerkleTree, OptimalMerkleTree, ZerokitMerkleTree};
+
+pub fn get_leaves(n: u32) -> Vec<Fr> {
+    (0..n).map(|s| Fr::from(s)).collect()
+}
+
+pub fn optimal_merkle_tree_poseidon_benchmark(c: &mut Criterion) {
+    c.bench_function("OptimalMerkleTree::full_height_gen", |b| {
```
Comment (rymnc) [inline rln/benches/poseidon_tree_benchmark.rs:13]:
```suggestion
    c.bench_function("OptimalMerkleTree::<Poseidon>::full_height_gen", |b| {
```
to differentiate between the benches in the utils crate (they use keccak, this one uses poseidon.)

File: utils/tests/merkle_tree.rs [hunk 3]
```diff
@@ -185,8 +170,8 @@ pub mod test {
         .unwrap();
 
         // ensure that the leaves are set correctly
-        for i in 0..new_leaves.len() {
-            assert_eq!(tree.get_leaf(i), new_leaves[i]);
+        for (i, &new_leave) in new_leaves.iter().enumerate() {
+            assert_eq!(tree.get_leaf(i), new_leave);
```
Comment (rymnc) [inline utils/tests/merkle_tree.rs:174]:
```suggestion
        for (i, &new_leaf) in new_leaves.iter().enumerate() {
            assert_eq!(tree.get_leaf(i), new_leaf);
```

**Review by rymnc — COMMENTED** [review]
LGTM in general, small changes inline :)
thanks for reducing the code surface!

**Review by rymnc — APPROVED** [review]
LGTM, thanks!

**Pull request merged**